### PR TITLE
Conditionally hide the language bar

### PR DESCRIPTION
**Pull request by coreyfloyd**
Waiting on asset from @nirzar so don't merge yet.

https://phabricator.wikimedia.org/T124094
https://phabricator.wikimedia.org/T124353

Adds setting for toggling the language bar on and off
Uses the number of selected languages in the welcome screen to set the initial value

### CODE REVIEW

**Comment by bgerstle** [comment]
@coreyfloyd FYI needs conflicts resolved

File: Wikipedia/Code/WMFSearchViewController.m [hunk 1]
```diff
@@ -172,31 +175,50 @@ - (void)configureSearchField {
...
+        [self selectLanguageForSite:self.searchSite];
+
+    }else{
+        [self.languageBarContainer removeFromSuperview];
+        [self.searchContentContainer mas_makeConstraints:^(MASConstraintMaker *make) {
+            make.top.equalTo(self.searchFieldContainer.mas_bottom);
+        }];
+    }
+    
+    }
```
Comment (bgerstle) [inline Wikipedia/Code/WMFSearchViewController.m]: nit: run uncrustify to fix this indentation

**Comment by bgerstle** [comment]
:-1: language bar is hidden when a I "upgraded" with previously selected preferred languages:

1. Checkout master
2. Uninstall app
3. Install app from master
4. Select another language or two
5. Checkout this PR
6. "Upgrade" the previous install by running from this PR
7. :x: Language bar is hidden despite having multiple preferred languages

**Comment by bgerstle** [comment]
Confirmed, behaves as expected for upgrade from 4.1.7 :white_check_mark:

**Comment by bgerstle** [comment]
Confirmed works for clean install w/ multiple preferred OS languages :white_check_mark:

**Comment by bgerstle** [comment]
:+1: lgtm, but uncrustify would be nice

**Comment by bgerstle** [comment]
@coreyfloyd left a [comment on the task](https://phabricator.wikimedia.org/T124094#2036731), but I'm assuming it's intentional that the language bar remains hidden even after a second site is selected?

**Comment by coreyfloyd** [comment]
@bgerstle done

**Comment by bgerstle** [comment]
:+1:

**Comment by nirzar** [comment]
@coreyfloyd oops https://phabricator.wikimedia.org/T124353#2038102 here you go...

**Comment by coreyfloyd** [comment]
@bgerstle @montehurd ready!

**Comment by coreyfloyd** [comment]
@bgerstle on another note - I think you are right about my files not getting uncrustified when I build without saving a file manually… will fix it so that doesn't happen.

**Comment by bgerstle** [comment]
:+1:'d already, @montehurd can you please review when you get in?

**Comment by montehurd** [comment]
Checking...

**Comment by montehurd** [comment]
@coreyfloyd @bgerstle hey guys @jtminor said he wants to think about one issue with this before we merge. 

Specifically, he's considering showing the lang bar by default if the device has more than preferred lang. Came up because when I was testing, on my sim with EN and HE as preferred langs, the bar disappeared. 

He says he'll either give go ahead to merge and/or file any change as a follow-on but he needs some time to think about it.

**Comment by bgerstle** [comment]
> Specifically, he's considering showing the lang bar by default if the device has more than preferred lang. Came up because when I was testing, on my sim with EN and HE as preferred langs, the bar disappeared.

i tested this, and it showed the language bar when I had more than one preferred language.  @montehurd you might have seen a false negative due to Xcode scheme settings messing with the OS settings again. I ran into this as well, and still plan to fix that later.  If you want to retest, you can either:

- Disable the "Run" arguments for the Wikipedia scheme
- Set the third one (AppleLanguages) to: `-AppleLanguages (en,he,fr)` or any other language codes you want set as preferred languages

**Comment by montehurd** [comment]
@bgerstle ok thanks! @jtminor said merge for now then we can address anything that comes up as follow-on.

**Pull request merged**